APIs/types.raml

```diff
@@ -0,0 +1,90 @@
+#%RAML 1.0 Types
+TokenAuthorizeRequest:
+  properties:
+    grant_type:
+      description: The OAuth 2.0 grant type of the request
+      type: string
+      required: true
+    code:
+      description:  The authorization code received from the authorization server
+      type: string
+      required: true
+    redirect_uri:
+      description: Absolute URI used by the auth server to redirect the user-agent, previously established during client registration
+      type: string
+    client_id:
+      description: The identifier of the client authenticating
+      type: string
+
+TokenPasswordRequest:
+  properties:
+    grant_type:
+      description: The OAuth 2.0 grant type of the request
+      type: string
+      required: true
+    username:
+      description: Username for the Password Credentials Grant
+      type: string
+      required: true
+    password:
+      description: Password for the Password Credentials grant_type
+      type: string
+      required: true
+    scope:
+      description: The scope of the access request
+      type: string
+      required: true
+
+TokenRefreshRequest:
+  properties:
+      grant_type:
+        description: MUST be set to "refresh_token"
+        required: true
+      refresh_token:
+        description: The refresh token issued to the client inside the bearer token
+        required: true
+      scope:
+        description: The scope of the access request
+        required: false
+
+RegisterClientRequest:
+  properties:
+    client_name:
+      description: Name of Client
+      type: string
+    client_uri:
+      description: URI of client
+      type: string
+    scope:
+      description: List of possible scopes, *space-separated*
+      type: string
+    redirect_uri:
+      description: Used by the authorization server to return responses containing authorization credentials to the client
+      type: array
+      required: true
+    grant_types:
+      description: The type of permissable OAuth 2.0 grants
+      type: array
+      items:
+        enum: [ authorization_code, implicit, password, client_credentials, refresh_token, urn:ietf:params:oauth:grant-type:jwt-bearer ]
```

Comment:
Looks like at least the URN will need quoting, otherwise it seems to throw some YAML parsing errors.